lib/portlayer/exec/container.go

```diff
@@ -377,15 +382,23 @@ func (c *Container) Signal(ctx context.Context, num int64) error {
 	return c.startGuestProgram(ctx, "kill", fmt.Sprintf("%d", num))
 }
 
-func (c *Container) LogReader(ctx context.Context) (io.ReadCloser, error) {
+func (c *Container) stopped() {
```

Comment:
Right, I added to call there to handle out-of-band power off and container shutting down on its own.  The Container.stop() method has an explicit call to Container.onStop to handle client stop call, in which case there's no need to check the state.